core/src/banking_stage.rs

```diff
@@ -1013,22 +1229,18 @@ impl BankingStage {
         tracer_packet_stats: &mut TracerPacketStats,
         bank_forks: &Arc<RwLock<BankForks>>,
     ) {
-        if let ForwardOption::NotForward = forward_option {
-            if !hold {
-                buffered_packet_batches.clear();
-            }
-            return;
-        }
+        let forward_option = unprocessed_transaction_storage.forward_option();
+        // The gossip thread should not even reach this point
+        assert!(!matches!(forward_option, ForwardOption::NotForward));
 
         // get current root bank from bank_forks, use it to sanitize transaction and
         // load all accounts from address loader;
         let current_bank = bank_forks.read().unwrap().root_bank();
         let mut forward_packet_batches_by_accounts =
             ForwardPacketBatchesByAccounts::new_with_default_batch_limits(current_bank);
-        let filter_forwarding_result = Self::filter_valid_packets_for_forwarding(
-            buffered_packet_batches,
-            &mut forward_packet_batches_by_accounts,
-        );
+        let filter_forwarding_result = unprocessed_transaction_storage
+            .filter_forwardable_packets_and_add_batches(&mut forward_packet_batches_by_accounts);
+
```

Comment:
The existing logic forwards up to the CU limit, prioritizing transactions that have a higher fee-per-cu weight.

I think the votes should have a separate forwarding logic to prioritize by stake weight up to the same CU limit, since votes do not have a per-cu fee

Reply:
This can be done in a separate PR though, feel free to create an issue.